LogicLayer/Notifications/MailSender.cs

```diff
@@ -0,0 +1,115 @@
+        }
+
+        public Task SendEventCancellationAsync(IEnumerable<string> emails, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.EventCancellationHtml),
+                link,
+                "Event Cancellation");
+            return SendEmailAsync(emails, notification);
+        }
+
+        public Task SendEventConfirmationAsync(IEnumerable<string> emails, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.EventConfirmationHtml),
+                link,
+                "Event Confirmation");
+            return SendEmailAsync(emails, notification);
+        }
+
+        public Task SendEventInvitationAsync(IEnumerable<string> emails, string link)
+        {
+            var notification = new Notification(GetHtmlPath(MailOptions.EventInvitationHtml), link, "Event Invitation");
+            return SendEmailAsync(emails, notification);
+        }
+
+        public Task SendFriendAcceptAsync(string email, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.FriendAcceptHtml),
+                link,
+                "Friend Invitation Accepted");
+            return SendEmailAsync(email, notification);
+        }
+
+        public Task SendFriendInvitationAsync(string email, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.FriendInvitationHtml),
+                link,
+                "Friend Invitation");
+            return SendEmailAsync(email, notification);
+        }
+
+        public Task SendPasswordResetAsync(string email, string link)
+        {
+            var notification = new Notification(GetHtmlPath(MailOptions.PasswordResetHtml), link, "Password Reset");
+            return SendEmailAsync(email, notification);
+        }
+
+        private Task SendEmailAsync(IEnumerable<string> emails, Notification notification)
+            => Execute(MailOptions.SendGridApiKey, emails, notification);
+
+        private Task SendEmailAsync(string email, Notification notification)
+        {
+            var emails = new List<string>
+            {
+                email
+            };
+            return SendEmailAsync(emails, notification);
+        }
+
+        private Task Execute(string apiKey, IEnumerable<string> emails, Notification notification)
+        {
+            var client = new SendGridClient(apiKey);
+            var from = new EmailAddress("GameBoard.contact@gmail.com", "GameBoard");
+            var tos = new List<EmailAddress>();
+            var subject = notification.Subject;
+            var htmlContent = notification.Html;
+
+            foreach (var email in emails)
+            {
+                tos.Add(new EmailAddress(email));
+            }
+
+            var msg = MailHelper.CreateSingleEmailToMultipleRecipients(from, tos, subject, null, htmlContent);
```

Comment:
Is it OK that we're using `ToMultipleRecipients` even for a single recipient? What are the differences between this and the alternatives?

Reply:
We can add another function, that just uses ToSingleRecipient and does everything else in the same way, that's the alternative. I'm pretty sure performance difference is irrelevant, we either use this or duplicate code, or make an inner function that does preprocessing and two functions that do Preprocess(); SendToMultiple/SendToSingle(). Whatever you prefer.

Reply:
Since you said that `ToSingleRecipient` works exactly the same as `ToMultipleRecipients`, then there's no reason to not keep this as it is.